src/browser-server/sagas/frontend-command-handlers.js

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import { takeEvery, call, select } from 'redux-saga/effects';
+
+import Connection from '../connection';
+import SharedActions from '../../shared/actions/shared-actions';
+import * as FrontendConnectionsSelectors from '../selectors/frontend-connections-selectors';
+
+function* registerKeyShortcuts({ meta: frontendConn, payload: shortcuts }) {
+  const frontendConnId = frontendConn.id;
+  const { runnerConnId, winId } = yield select(FrontendConnectionsSelectors.getFrontendClientMetaData, frontendConnId);
+
+  const action = SharedActions.commands.fromServer.toRunner.app.window.keyShortcuts.register({
+    winId,
+    shortcuts,
+  });
+
+  const runnerConn = Connection.getWithId(runnerConnId);
```

Comment:
I'd expect we would be using `FrontendConnection.getWithFrontendWinId` here instead of `Connection.getWithId` (ditto for all the other places throughout browser-server that take in a frontendConn).  Or, if there's not additional value being added by handling these connections as FrontendConnections then maybe we don't need the new class at all and instead can make the `frontendWinId` an optional property on the plain `Connection` object (and add the `static getWithFrontendWinId` there as well) then just always deal with connections.